src/dissect/dissect.c

```diff
@@ -823,7 +841,82 @@ static int action_copy(DissectedImage *m, LoopDevice *d) {
         return 0;
 }
 
+static int action_umount(const char *path) {
+        _cleanup_free_ char *absolute = NULL;
+        dev_t devno;
+        const char *node;
+        _cleanup_(loop_device_unrefp) LoopDevice *d = NULL;
+        _cleanup_(sd_device_unrefp) sd_device *device = NULL;
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *part;
+        int r;
+
+        r = path_make_absolute_cwd(path, &absolute);
```

Comment:
hmm, shouldn't this rather be chase_symlinks()? i.e. when operating with mounts that's usually the best approach.

And then maybe even use chase_symlinks_and_open() with O_DIRECTORY. and then use fd_is_mount_point() on the returned fd, as well as fd_get_whole_disk() (i.e. go via fd, rather than path more often then not). Of course, the fd needs be closed before any umount is attempted...